src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/TagParser.java

```diff
@@ -190,8 +190,7 @@ private static String getTagId(String[] javadocText, Point tagStart) {
             // Character.isJavaIdentifier... may not be a valid HTML
             // identifier but is valid for generics
             while (position < text.length()
-                    && (Character.isJavaIdentifierStart(text.charAt(position))
-                        || Character.isJavaIdentifierPart(text.charAt(position)))) {
+                    && Character.isJavaIdentifierPart(text.charAt(position))) {
```

Comment:
See https://docs.oracle.com/javase/specs/jls/se19/html/jls-3.html#jls-Identifier

Characters that satisfy `Character#isJavaIdentifierStart` are a subset of those that satisfy `Character.isJavaIdentifierPart`.  We can try to create a test with the following to kill this mutation:
```
➜  src cat Test.java  
public class Test {
    /**
     * @2x
     */
    void m(){}

}
➜  src javac Test.java  
➜  src javadoc Test.java
Loading source file Test.java...
Constructing Javadoc information...
Building index for all the packages and classes...
Standard Doclet version 21+35-LTS
Building tree for all the packages and classes...
Generating ./Test.html...
Test.java:1: warning: no comment
public class Test {
       ^
Test.java:1: warning: use of default constructor, which does not provide a comment
public class Test {
       ^
Generating ./package-summary.html...
Generating ./package-tree.html...
Generating ./overview-tree.html...
Building index for all classes...
Generating ./allclasses-index.html...
Generating ./allpackages-index.html...
Generating ./index-all.html...
Generating ./search.html...
Generating ./index.html...
Generating ./help-doc.html...
2 warnings


```

Reply:
yes done, but sadly it doesn't look like it killed any mutation :(

Reply:
Ah, I can see that I misread the diff. I thought we were removing ...part... and keeping ...start...

Ok, this makes sense to remove this condition for the reasons stated at https://github.com/checkstyle/checkstyle/pull/14189#discussion_r1438610590, namely that characters that satisfy isJavaIdentifierStart are a subset of the characters that satisfy isJavaIdentifierStart.

Reply:
@suniti0804 please restore your removal of this condition

Reply:
Done